pkg/gofr/gofr_test.go

```diff
@@ -506,35 +506,16 @@ func Test_AddCronJob_Success(t *testing.T) {
 }
 
 func TestStaticHandler(t *testing.T) {
-	// Generate a public directory endpoint
-	directory := "./public"
-
-	if _, err := os.Stat(directory); err != nil {
-		err := os.Mkdir("./public", 0755)
-		if err != nil {
-			t.Errorf("Couldn't create a public directory, error: %s", err)
-		}
-	}
-	// Application Generation
-	app := New()
-
 	// Generate some files for testing
 	htmlContent := []byte("<html><head><title>Test Static File</title></head><body><p>Testing Static File</p></body></html>")
 
-	file1, errFile := os.Create("./public/index.html")
-
-	if errFile != nil {
-		t.Error("Couldn't create index.html file")
+	err := createPublicDirectory(t, htmlContent)
+	if err != nil {
+		return
```

Comment:
to respect the pattern you are using in your test, you should use
	
```go
err := createPublicDirectory(t, htmlContent)
if err != nil {
	t.Fatalf(err)
}
```

and do not log error in createPublicDirectory (except the cleanup you can do nothing except log)